templateutils/templateutils.go

```diff
@@ -108,6 +118,154 @@ var funcMap = template.FuncMap{
+	},
+	reflect.Uint: func(v1, v2 interface{}) bool {
+		return v1.(uint) < v2.(uint)
+	},
+	reflect.Uint8: func(v1, v2 interface{}) bool {
+		return v1.(uint8) < v2.(uint8)
+	},
+	reflect.Uint16: func(v1, v2 interface{}) bool {
+		return v1.(uint16) < v2.(uint16)
+	},
+	reflect.Uint32: func(v1, v2 interface{}) bool {
+		return v1.(uint32) < v2.(uint32)
+	},
+	reflect.Uint64: func(v1, v2 interface{}) bool {
+		return v1.(uint64) < v2.(uint64)
+	},
+	reflect.Float64: func(v1, v2 interface{}) bool {
+		return v1.(float64) < v2.(float64)
+	},
+	reflect.Float32: func(v1, v2 interface{}) bool {
+		return v1.(float32) < v2.(float32)
+	},
+	reflect.String: func(v1, v2 interface{}) bool {
+		return v1.(string) < v2.(string)
+	},
+}
+
+var sortDscMap = map[reflect.Kind]func(interface{}, interface{}) bool{
+	reflect.Int: func(v1, v2 interface{}) bool {
+		return v2.(int) < v1.(int)
+	},
+	reflect.Int8: func(v1, v2 interface{}) bool {
+		return v2.(int8) < v1.(int8)
+	},
+	reflect.Int16: func(v1, v2 interface{}) bool {
+		return v2.(int16) < v1.(int16)
+	},
+	reflect.Int32: func(v1, v2 interface{}) bool {
+		return v2.(int32) < v1.(int32)
+	},
+	reflect.Int64: func(v1, v2 interface{}) bool {
+		return v2.(int64) < v1.(int64)
+	},
+	reflect.Uint: func(v1, v2 interface{}) bool {
+		return v2.(uint) < v1.(uint)
+	},
+	reflect.Uint8: func(v1, v2 interface{}) bool {
+		return v2.(uint8) < v1.(uint8)
+	},
+	reflect.Uint16: func(v1, v2 interface{}) bool {
+		return v2.(uint16) < v1.(uint16)
+	},
+	reflect.Uint32: func(v1, v2 interface{}) bool {
+		return v2.(uint32) < v1.(uint32)
+	},
+	reflect.Uint64: func(v1, v2 interface{}) bool {
+		return v2.(uint64) < v1.(uint64)
+	},
+	reflect.Float64: func(v1, v2 interface{}) bool {
+		return v2.(float64) < v1.(float64)
+	},
+	reflect.Float32: func(v1, v2 interface{}) bool {
+		return v2.(float32) < v1.(float32)
+	},
+	reflect.String: func(v1, v2 interface{}) bool {
+		return v2.(string) < v1.(string)
+	},
+}
+
+type valueSorter struct {
+	val   reflect.Value
+	field int
+	less  func(v1, v2 interface{}) bool
+}
+
+func (v *valueSorter) Len() int {
+	return v.val.Len()
+}
+
+func (v *valueSorter) Less(i, j int) bool {
+	iVal := v.val.Index(i)
+	jVal := v.val.Index(j)
+	return v.less(iVal.Field(v.field).Interface(), jVal.Field(v.field).Interface())
+}
+
+func (v *valueSorter) Swap(i, j int) {
+	iVal := v.val.Index(i).Interface()
+	jVal := v.val.Index(j).Interface()
+	v.val.Index(i).Set(reflect.ValueOf(jVal))
+	v.val.Index(j).Set(reflect.ValueOf(iVal))
+}
+
+func newValueSorter(obj interface{}, field string, ascending bool) *valueSorter {
+	val := reflect.ValueOf(obj)
+	typ := reflect.TypeOf(obj)
+	sTyp := typ.Elem()
+
+	var index int
+	var fTyp reflect.StructField
+	for index = 0; index < sTyp.NumField(); index++ {
+		fTyp = sTyp.Field(index)
+		if fTyp.Name == field {
+			break
+		}
+	}
+	if index == sTyp.NumField() {
+		panic(fmt.Sprintf("%s is not a valid field name", field))
+	}
+	fKind := fTyp.Type.Kind()
```

Comment:
Do you also need to handle pointers here? So that pointers to integers do the right thing?